compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/SymbolTable.java

```diff
@@ -517,6 +519,35 @@ public void defineOperators() {
 
     }
 
+    private void defineXmlArithmaticOperations() {
+        defineBinaryOperator(OperatorKind.ADD, xmlType, xmlType, xmlType);
+        List<BType> xmlSubTypes = Arrays.asList(xmlElementType, xmlCommentType, xmlPIType, xmlTextType);
+        for (int i = 0; i < xmlSubTypes.size(); i++) {
+            BType subType = xmlSubTypes.get(i);
+            defineBinaryOperator(OperatorKind.ADD, subType, subType, subType);
+
+            List<BType> others = new ArrayList<>(xmlSubTypes);
+            others.remove(i);
```

Comment:
Do we need this new list? Can simply use 2 nested loops to achieve this right? The basic idea would be to use 
```java
defineBinaryOperator(OperatorKind.ADD, subType, subType, subType);
```
for when `i == j` and use 
```java
defineBinaryOperator(OperatorKind.ADD, subType, otherType, xmlType);
defineBinaryOperator(OperatorKind.ADD, otherType, subType, xmlType);
```
when `i != j`

Reply:
+1 made removed the list.